### PR TITLE
[r] Index hca_dev_20201023_ebiv4___20210302 into dcp2ebi on dev (#2851)

### PR DESCRIPTION
**Pull request by amarjandu**
Author

- [x] PR title references issue
- [x] Title of main commit references issue
- [x] PR is linked to Zenhub issue

Author (reindex)

- [x] Added `r` tag to commit title                         <sub>or this PR does not require reindexing</sub>
- [x] Added `reindex` label to PR                           <sub>or this PR does not require reindexing</sub>

Author (freebies & chains)

- [x] Freebies are blocked on this PR                       <sub>or there are no freebies in this PR</sub>
- [x] Freebies are referenced in commit titles              <sub>or there are no freebies in this PR</sub>
- [x] This PR is blocked by previous PR in the chain        <sub>or this PR is not chained to another PR</sub>
- [x] Added `chain` label to the blocking PR                <sub>or this PR is not chained to another PR</sub>

Author (upgrading)

- [x] Documented upgrading of deployments in UPGRADING.rst  <sub>or this PR does not require upgrading</sub>
- [x] Added `u` tag to commit title                         <sub>or this PR does not require upgrading</sub>
- [x] Added `upgrade` label to PR                           <sub>or this PR does not require upgrading</sub>

Author (requirements, before every review)

- [x] Ran `make requirements_update`                        <sub>or this PR leaves requirements*.txt, common.mk and Makefile untouched</sub>
- [x] Added `R` tag to commit title                         <sub>or this PR leaves requirements*.txt untouched</sub>
- [x] Added `reqs` label to PR                              <sub>or this PR leaves requirements*.txt untouched</sub>

Author (before every review)

- [x] `make integration_test` passes in personal deployment <sub>or this PR does not touch functionality that could break the IT</sub>
- [x] Rebased branch on `develop`, squashed old fixups

Primary reviewer (after approval)

- [x] Commented on demo expectations                        <sub>or labelled PR as `no demo`</sub>
- [x] Decided if PR can be labeled `no sandbox`
- [x] PR title is appropriate as title of merge commit
- [x] Moved ticket to Approved column
- [x] Assigned PR to an operator

Operator (before pushing merge the commit)

- [x] Checked `reindex` label and `r` commit title tag
- [x] Checked that Demo expectations are clear              <sub>or PR is labeled as `no demo`</sub>
- [x] Rebased and squashed branch
- [x] Sanity-checked history
- [x] Pushed PR branch to Github
- [x] Branch pushed to Gitlab and build passes in `sandbox` <sub>or added `no sandbox` label</sub>
- [x] Started reindex in `sandbox`                          <sub>or this PR does not require reindexing `sandbox`</sub>
- [x] Checked for failures in `sandbox`                     <sub>or this PR does not require reindexing `sandbox`</sub>
- [x] Added PR reference to merge commit title
- [x] Collected commit title tags in merge commit title
- [x] Moved linked issue to Merged column
- [ ] Pushed merge commit to Github

Operator (after pushing the merge commit)

- [ ] Moved freebies to Merged column                       <sub>or there are no freebies in this PR</sub> 
- [ ] Shortened the PR chain                                <sub>or this PR is not the base of another PR</sub>
- [ ] Verified that `N reviews` labelling is accurate
- [ ] Pushed merge commit to Gitlab                         <sub>or this changes can be pushed later, together with another PR</sub>
- [ ] Deleted PR branch from Github and Gitlab

Operator (reindex) 

- [ ] Purchased BQ slot committment in `dev`                <sub>or this PR does not require reindexing `dev`</sub>
- [ ] Started reindex in `dev`                              <sub>or this PR does not require reindexing `dev`</sub>
- [ ] Deleted BQ slot committment in `dev`                  <sub>or this PR does not require reindexing `dev`</sub>
- [ ] Checked for failures in `dev`                         <sub>or this PR does not require reindexing `dev`</sub>
- [ ] Purchased BQ slot committment in `prod`               <sub>or this PR does not require reindexing `prod`</sub>
- [ ] Started reindex in `prod`                             <sub>or this PR does not require reindexing `prod`</sub>
- [ ] Deleted BQ slot committment in `prod`                 <sub>or this PR does not require reindexing `prod`</sub>
- [ ] Checked for failures in `prod`                        <sub>or this PR does not require reindexing `prod`</sub>

Operator

- [ ] Unassigned PR

### CODE REVIEW

**Comment by amarjandu** [comment]
looks like IT fails. https://gitlab.dev.singlecell.gi.ucsc.edu/ucsc/azul/-/jobs/11364
Nothing the queues, going to retry IT.

**Comment by amarjandu** [comment]
IT passed after being ran again, see https://github.com/DataBiosphere/azul/issues/2851#issuecomment-790182938 for information on error.

**Comment by amarjandu** [comment]
Starting reindex on sandbox, verified that failed queues are empty. 
```
(.venv) Specter:azul amar$ python scripts/manage_queues.py list
2021-03-03 16:29:22,381 INFO    MainThread: Listing queues

Queue Name                          Messages Available  Messages In Flight  Messages Delayed

2021-03-03 16:29:22,424 INFO    MainThread: Found credentials in environment variables.
azul-notifications-sandbox                  0                   0                  0
azul-tallies-sandbox.fifo                   0                   0                  0
azul-tallies_retry-sandbox.fifo             0                   0                  0
azul-tallies_fail-sandbox.fifo              0                   0                  0
azul-notifications_fail-sandbox             0                   0                  0
```

**Comment by amarjandu** [comment]
reindex on `sandbox` passed without failures.

File: deployments/dev/environment.py [hunk 1]
```diff
@@ -49,6 +49,7 @@ def env() -> Mapping[str, Optional[str]]:
             f'AZUL_TDR_{catalog.upper()}_SOURCES': ','.join([
                 'tdr:broad-jade-dev-data:snapshot/hca_dev_20210208_snaptest2___20210208',
                 'tdr:broad-jade-dev-data:snapshot/hca_dev_20210211_snaptest3___20210211',
```
Comment (hannes-ucsc) [inline deployments/dev/environment.py:51]: These should be removed. I should have specified that we want just this one snapshot in the catalog.

File: deployments/sandbox/environment.py [hunk 2]
```diff
@@ -59,6 +59,7 @@ def env() -> Mapping[str, Optional[str]]:
             f'AZUL_TDR_{catalog.upper()}_SOURCES': ','.join([
                 'tdr:broad-jade-dev-data:snapshot/hca_dev_20210208_snaptest2___20210208',
                 'tdr:broad-jade-dev-data:snapshot/hca_dev_20210211_snaptest3___20210211',
```
Comment (hannes-ucsc) [inline deployments/sandbox/environment.py:61]: Same here

**Review by hannes-ucsc — CHANGES_REQUESTED** [review]
We

**Review by hannes-ucsc — APPROVED** [review]

**Comment by amarjandu** [comment]
There are 4 failures notifications in the snapshot hca_dev_20201203___20201210 that were observed in the develop failed queue. These are known failures. https://github.com/DataBiosphere/azul/wiki/Operator-Manual#notes

Going to purge the queues then start the reindex of the `dcp2ebi` catalog.

**Comment by amarjandu** [comment]
reindex of `dev` is complete, no failed notifications.

reindexed with:
`python scripts/reindex.py --delete --index --purge --partition-prefix-length=2 --catalog dcp2ebi`

**Pull request merged**